### ISSUE TITLE
[Feature request] Do not suspend tab when text in forms are not yet submitted

### ISSUE THREAD

**Issue by ReporterX** [k5md/Total-Suspender-webextension#6]
Do not suspend a tab when text in forms are not yet submitted.

It is useful when you are filling in forms, or typing in forums etc. 

This is pretty important. Thank you.

**Comment by k5md** [comment]
Hello, @ReporterX !
To be honest, I can not reproduce the issue with data in forms disappearing, the extension uses native discarding, so the data is always preserved for those forms, that do not use fetched suggestions.
I may be mistaken, so could you, please, provide an example of a site with this bug appearing?

**Comment by ReporterX** [comment]
Not all text in any form field is preserved. Some don't.

Simply open a new issue here.  
https://github.com/k5md/Total-Suspender-webextension/issues/new

Type something in the title bar.

Discard the tab.

Text typed in the title bar will be gone.

**Comment by k5md** [comment]
I suppose, this issue is more firefox-related, I've tried to reproduce this without the extension installed to check, wheter or not it's related to it. Without extension, when closing and then opening the browser with tabs with some data in forms it's the same - some data is lost and some is preserved, so, I guess, this is an issue with firefox not storing all data to cache on discard.
If it is so, to fix it the extension will need to directly access every page's content and store form data, then load it. To be honest, in this case I do not want to interfere with users' tabs content in any way for these reasons:
- Soon there may be a bugfix for it
- It could make it less secure
- It's more suitable for a separate webextension

I'll try to find more info on it, maybe I'm mistaken, search for existing extensions, that fix it and if there is none, maybe write my own.

**Comment by ReporterX** [comment]
It might be because Firefox supports caching data from <textarea> tag or its similar only. It is easy for humans to pinpoint a form field, but it seems there are diverse ways (codes) to display form fields technically. At least we should catch the common ones.

For your information, [Auto Tab Discard](https://addons.mozilla.org/en-US/firefox/addon/auto-tab-discard/) has this option. It is called "Do not discard a tab when form changes are not yet submitted". I prefer your addon. Yours is more reliable. I like your interface.

**Comment by k5md** [comment]
I see now, for some reason I've thought it's necessary to fix that behaviour, but it's actually not! If I introduce an option to ignore tabs with not submitted form data, there would be no need in storing/loading user's form data, the extension should only detect it. Got it, I will implement this feature soon.

**Comment by ReporterX** [comment]
Yep just detect it. I believe it is easier to implement than trying to fix the behavior. Thank you for writing such a good addon.

**Comment by xpuu** [comment]
[Great Suspender](https://chrome.google.com/webstore/detail/the-great-suspender/klbibkeccnjlkjkiokjodocebajanakg?hl=en) also has this feature. 
> Detects tabs that contain forms with user input and can prevent them from suspending

It's pretty annoying if it closes the form after half an hour of work e.g. in Confluence, Jira etc. But of course, thanks for the add on, apart of this little thing, it works great!

**Comment by reenberg** [comment]
Not discarding tabs with active data is the one reason for me stopping to use this addon.  Gitlab has lots of fields that loses data if you are reviewing a merge request, or writing comments on issues.
